src/components/chat/item-header.tsx

```diff
@@ -7,16 +7,33 @@ interface Props {
   onClickEdit: (e: React.MouseEvent<HTMLAnchorElement>) => void;
   onClickReply: (e: React.MouseEvent<HTMLAnchorElement>) => void;
   onRemove: (chat: Chat) => void;
+  collapsed: boolean;
+  setCollapsed: (collapsed: boolean) => void;
 }
 
 export default function ChatItemHeader({
   chat,
   onClickEdit,
   onClickReply,
   onRemove,
+  collapsed,
+  setCollapsed,
 }: Props) {
   return (
     <div className="header">
+      <span
+        onClick={() => {
+          if (chat.children) {
+            if (chat.children.length > 0) setCollapsed(!collapsed);
```

Comment:
When we need the previous state value to calculate the new one, we should do a functional update :wink: